libcloudforensics/providers/azure/internal/common.py

```diff
@@ -34,21 +35,103 @@
+              'subscriptionId': yyy,
+              'tenantId': yyy,
+              'clientId': yyy,
+              'clientSecret': yyy
+          },
+          ...
+        }
+
+        Note that you can specify several profiles that use the same tenantId,
+        clientId and clientSecret but a different subscriptionId.
+        If you set the environment variable AZURE_CREDENTIALS_PATH to an
+        absolute path to the credentials file, then the library will look
+        there instead of in ~/.azure/credentials.json.
+
   Returns:
-    ServicePrincipalCredentials: Azure credentials.
+    Tuple[str, ServicePrincipalCredentials]: Subscription ID and
+        corresponding Azure credentials.
+
+  Raises:
+    RuntimeError: If the credential file is not found.
+    ValueError: If the requested profile name is not found in the credential
+        file or if there are missing entries in the profile name.
   """
-  return ServicePrincipalCredentials(tenant=os.environ["AZURE_TENANT_ID"],
-                                     client_id=os.environ["AZURE_CLIENT_ID"],
-                                     secret=os.environ["AZURE_CLIENT_SECRET"])
+  # pylint: enable=line-too-long
+  if not profile_name:
+    subscription_id = os.getenv('AZURE_SUBSCRIPTION_ID')
+    client_id = os.getenv("AZURE_CLIENT_ID")
+    secret = os.getenv("AZURE_CLIENT_SECRET")
+    tenant = os.getenv("AZURE_TENANT_ID")
+    if not (subscription_id and client_id and secret and tenant):
+      raise RuntimeError('Please make sure you defined the following '
+                         'environment variables: [AZURE_SUBSCRIPTION_ID,'
+                         'AZURE_CLIENT_ID, AZURE_CLIENT_SECRET,'
+                         'AZURE_TENANT_ID].')
+    return subscription_id, ServicePrincipalCredentials(client_id,
+                                                        secret,
+                                                        tenant=tenant)
+
+  path = os.getenv('AZURE_CREDENTIALS_PATH')
+  if not path:
+    path = os.path.expanduser('~/.azure/credentials.json')
+
+  if not os.path.exists(path):
+    raise RuntimeError('Credential files not found. Please place it in '
+                       '"~/.azure/credentials.json" or specify an absolute '
+                       'path to it in the AZURE_CREDENTIALS_PATH environment '
+                       'variable.')
+
+  with open(path) as profiles:
+    try:
+      account_info = json.load(profiles).get(profile_name)
+    except ValueError as exception:
+      raise ValueError('Could not decode JSON file. Please verify the file '
+                       'format: {0:s}'.format(str(exception)))
+    if not account_info:
+      raise ValueError(
+          'Profile name {0:s} not found in credentials file {1:s}'.format(
+              profile_name, path))
+    required_entries = ['subscriptionId', 'clientId', 'clientSecret',
+                        'tenantId']
+    if not all(entry in account_info for entry in required_entries):
```

Comment:
I assume all of these must have values, right? In that case, you can do 
```suggestion
    if not all(account_info.get(entry) for entry in required_entries):
```

Reply:
Correct, changed